### PR TITLE
3 get categories

### PR DESCRIPTION
**Pull request by PoppyLew**
app, controller and model set up
test suite set up
all tests including error handling passed
ticket 3 complete

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+
+app.get('/api/categories', getCategories)
+
+app.all('/*', (req,res,next) => {
+  res.status(404).send({msg: 'Not Found'})
+})
+
+app.use((err, req, res, next) => {
+    console.log(err);
```
Comment (Poonam-raj) [inline app.js]: make sure you remove all console logs before making a PR

File: package.json [hunk 2]
```diff
@@ -21,9 +21,11 @@
   },
   "homepage": "https://github.com/northcoders/be-nc-games#readme",
   "dependencies": {
-    "pg-format": "^1.0.4",
     "dotenv": "^16.0.0",
-    "pg": "^8.7.3"
+    "express": "^4.18.1",
+    "pg": "^8.7.3",
+    "pg-format": "^1.0.4",
+    "supertest": "^6.2.4"
```
Comment (Poonam-raj) [inline package.json]: supertest should be a devDependency

File: package.json [hunk 3]
```diff
@@ -21,9 +21,11 @@
...
   },
   "devDependencies": {
     "husky": "^7.0.0",
```
Comment (Poonam-raj) [inline package.json:30]: jest should be in your devDependencies

File: __tests__/app.test.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+describe('get categories api', () => {
+    it('responds with all categories with correct keys', () => {
+        return request(app)
+      .get('/api/categories')
+      .expect(200)
+      .then(({ body }) => {
+        const {categories: categories} = body;
+        expect(categories).toBeInstanceOf(Array);
+        categories.forEach((category) => {
```
Comment (Poonam-raj) [inline __tests__/app.test.js:19]: whenever you so a forEach to check all the values in the array are correct you should also check the length of the returned array to make sure you're getting the right amount of data back

put that in and you're all good with these test

**Review by Poonam-raj — APPROVED** [review]
Looks good! Nice structure, sensible naming, only a few comments mainly on your dependencies and a missing check on your test. Good to merge 🎉

File: __tests__/app.test.js (hunk 4 above)
Comment (PoppyLew) [reply __tests__/app.test.js:19]: thank you for all the amazing feedback, issues resolved and merging now

**Pull request merged**